.github/workflows/quick_test.yml

```diff
@@ -0,0 +1,49 @@
+name: Quick Test
```

Comment:
Could you say more about why this is being added? We're now maintaining two separate GitHub Actions workflows for tests, which has some maintenance / complexity cost.

Reply:
It's not hugely needed, but it seemed slightly overkill to run the full matrix of tests on every push. This is just an identical copy of tests with change in matrix, which I thought it was relatively low complexity idea. But happy to remove and revert changes to the full tests.

Reply:
It's definitely a subjective call, but I'd prefer to keep the single workflow file for now. Certainly not opposed to breaking things out into multiple workflows if we're feeling development cycle pain. But my sense is currently it's probably makes more sense to keep things in one place. 

Otherwise, this PR LGTM